packages/front-end/components/Settings/DisplayTestQueryResults.tsx

```diff
@@ -0,0 +1,114 @@
+import React from "react";
+import { TestQueryRow } from "back-end/src/types/Integration";
+
+export type TestQueryResults = {
+  status: number;
+  optionalColumns?: string[];
+  duration?: string;
+  error?: string;
+  results?: TestQueryRow[];
+  includesNamedColumns: boolean;
+};
+
+type Props = {
+  testQueryResults: TestQueryResults | null;
+  // eslint-disable-next-line
+  form?: any;
```

Comment:
It looks like, based on the component, at least `"dimensions"` is required on the form. Are we able to type it with a form type, e.g. something like `UseFormReturn<{ dimensions: string[] }>` (or whatever the dimensions and other watched properties are)? There are other examples of this in the code base, so I imagine it's possible.

We are making an effort in the code base to not add more `any` if we can avoid it. This one looks like it's possible to avoid it but let me know if I'm overlooking something.